File: problem2.rb
def even_sum_fibonacci(n = 4000000)
  last = 1
  current = 2
  even_sum = 0
  while current < n do
    even_sum += current if current.even?
    temp = current
    current += last
    last = temp
  end
  even_sum
end

max_fib = ARGV.first ? ARGV.first.to_i : 10
puts even_sum_fibonacci(max_fib)

File: C#/problem2.cs
using System;
using System.Diagnostics;

class Problem2
{
  public static void Main(string[] args)
  {
    long maxFib = 10;
    if (args.Length > 0)
    {
      maxFib = Convert.ToInt64(args[0]);
    }
    Stopwatch stopwatch = new Stopwatch();
    stopwatch.Start();
    Console.WriteLine(EvenSumFibonacci(maxFib));
    stopwatch.Stop();
    Console.WriteLine(stopwatch.ElapsedMilliseconds / 1000.0 + " seconds");
  }

  private static long EvenSumFibonacci(long n)
  {
    long last = 1;
    long current = 2;
    long evenSum = 0;
    while (current < n)
    {
      if (current % 2 == 0)
      {
        evenSum += current;
      }
      long temp = current;
      current += last;
      last = temp;
    }
    return evenSum;
  }
}
